velox/common/caching/ScanTracker.cpp

```diff
@@ -21,6 +21,19 @@
 
 namespace facebook::velox::cache {
 
+TrackingData& ScanTracker::trackingData(TrackingId id) {
```

Comment:
This is not safe to access through an object reference within a mutable STL container.

What if we just use read lock for read access and write lock for write access? What's the additional improvement with this optimization? thanks!

Reply:
Using wlock to update tracking data would be similar to accessing with mutex, which blocks updates to non-colliding tracking IDs. I believe @Yuhta is trying to optimize updating TrackingData for distinct tracking ids.

Reply:
Ok. Can you clarify in the doc:
(1) use a single r/w lock to manage tracking data in scan spec
(2) each tracking data use a separate lock handle the update to the tracking data itself

And make the tracking data a shared ptr in the scan spec map.